src/Libraries/RevitNodes/Elements/Element.cs

```diff
@@ -327,14 +327,31 @@ public object GetParameterValueByName(string parameterName)
             switch (param.StorageType)
             {
                 case StorageType.ElementId:
-                    int id = param.AsElementId().IntegerValue;
-                    // When the element is obtained here, to convert it to our element wrapper, it
-                    // need to be figured out whether this element is created by us. Here the existing
-                    // element wrappers will be checked. If there is one, its property to specify
-                    // whether it is created by us will be followed. If there is none, it means the
-                    // element is not created by us.
-                    var ele = ElementIDLifecycleManager<int>.GetInstance().GetFirstWrapper(id) as Element;
-                    result = ElementSelector.ByElementId(id, ele == null ? true : ele.IsRevitOwned);
+                    int paramId = param.Id.IntegerValue;
+                    int valueId = param.AsElementId().IntegerValue;
+                    if (valueId > 0)
```

Comment:
I think this check shouldn't be based on -ve value of elementId, rather we should check for parameterId, and if the parameter belongs to Category, then we should return specific category.